drivers/hwinfo/hwinfo_andes.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2021 Andes Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/hwinfo.h>
+#include <string.h>
+#include <soc.h>
+
+ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
+{
+	uint32_t ver = sys_read32(SMU_BASE + SMU_SYSTEMVER) >> 8;
+	uint8_t id[3];
+
+	if (length > sizeof(id)) {
+		length = sizeof(id);
+	}
+
+	id[0] = (uint8_t)(ver >> 16);
+	id[1] = (uint8_t)(ver >> 8);
+	id[2] = (uint8_t)(ver >> 0);
+
+	memcpy(buffer, id, length);
+
+	return length;
+}
+
+int z_impl_hwinfo_get_reset_cause(uint32_t *cause)
+{
+	uint32_t flags = 0;
+
+	uint32_t reason = sys_read32(SMU_BASE + SMU_WRSR);
+
+	if (reason & SMU_WRSR_APOR) {
+		flags |= RESET_POR;
+	}
+	if (reason & SMU_WRSR_MPOR) {
+		flags |= RESET_POR;
+	}
+	if (reason & SMU_WRSR_HW) {
+		flags |= RESET_PIN;
+	}
+	if (reason & SMU_WRSR_WDT) {
+		flags |= RESET_WATCHDOG;
+	}
+	if (reason & SMU_WRSR_SW) {
+		flags |= RESET_SOFTWARE;
+	}
+
+	*cause = flags;
+
+	return 0;
+}
+
+int z_impl_hwinfo_clear_reset_cause(void)
+{
+	sys_write32(BIT_MASK(5), SMU_BASE + SMU_WRSR);
```

Comment:
Please give the bit 5 a name.
Otherwise looks good to me.
Thanks!

Reply:
Thanks for the review. Added it.